addons/account/models/account_move.py

```diff
@@ -2177,7 +2177,7 @@ def post(self):
                     raise UserError(_('Please define a sequence on your journal.'))
 
                 # Consume a new number.
-                to_write['name'] = sequence.next_by_id(sequence_date=move.date)
+                to_write['name'] = sequence.with_context(ir_sequence_date=move.date).next_by_id()
```

Comment:
It doesn't work because of:
https://github.com/odoo/odoo/blob/78ce8bc5a41af7852e6ed6a0d69ece57e3df958a/odoo/addons/base/models/ir_sequence.py#L251

I don't understand what you are trying to fix... Sequence date are indexed by year.

Reply:
You need to change ir_sequence itself to always use the `sequence_date` parameter instead of mismatching the dates from context and from parameters